### PR TITLE
Make check on "fileExtensions" and "excludeFiles" for specified path.

### PR DESCRIPTION
**Pull request by aliosv**

### CODE REVIEW

**Comment by mrjoelkemp** [comment]
Thanks @aliosv. Looks like a few tests broke with this PR.

**Comment by markelog** [comment]
@aliosv could you elaborate your issues and please add tests to this PR

**Comment by aliosv** [comment]
@markelog, pre-commit hook run this script https://github.com/bem/bem-components/blob/v2/.githooks/pre-commit/lint, but there is a problem: jscs will check all staged files, not taking into account own config.

**Comment by markelog** [comment]
Still don't understand the issue, you saying `excludeFiles` doesn't work?

Could provide tests for this pull? With them, it would be easier to grasp exactly what do you trying to fix.

**Comment by aliosv** [comment]
excludeFiles and fileExtesions doesn't work for specified path, i.e. ````jscs <path>```` will check the file regardless of the config.
I will add tests as soon as i can.

**Comment by aliosv** [comment]
added tests

File: test/specs/checker.js [hunk 1]
```diff
@@ -43,9 +43,29 @@ describe('modules/checker', function() {
...
+
+            it('should return empty array of errors for excluded file', function() {
+                checker.configure({ excludeFiles: ['./test/**'] });
+
+                return checker.checkPath('./test/data/checker/file.js').then(function(errors) {
+                    assert(errors.length === 0);
+                });
+            });
+
+            it('should return empty array of errors for file with not registered extension', function() {
```
Comment (markelog) [inline test/specs/checker.js]: Without changes in the source, only this test is falling whereas this behaviour is by design, we open to discuss this issue, but it seems point of this pull should be about another matter?

**Comment by markelog** [comment]
@aliosv still interested?

**Comment by aliosv** [comment]
@markelog, yes, but i do not see what to discuss. PR adds new feature: check specified path to file by config settings, it's broke backward compatibility, but you still have possibility to check any file using stdin. Maybe you know how to solve my issue in other way?

**Comment by markelog** [comment]
I don't understand what is the issue, tests your added are passing without source changes

**Comment by aliosv** [comment]
@markelog, https://github.com/bem/bem-components/blob/v2/.githooks/pre-commit/lint#L27
на пре-коммит хуке jscs может валидировать файлы, которые запрещены в его конфиге, этого не хочется.

**Comment by markelog** [comment]
Please use english, otherwise others can't participate in the discussion.

I'm afraid you are not to descriptive, link to the repo doesn't explain anything, please add failing tests to this pr or create a repo where possible bug could be reproduced.

**Comment by aliosv** [comment]
@markelog, test 'should return empty array of errors for excluded file' should fail.

**Comment by markelog** [comment]
So can you add a falling test?

**Comment by aliosv** [comment]
@markelog, https://github.com/aliosv/node-jscs-test project to test. Also i find my error: each file is checked on excluding, so problem only in cheking on allowed extension.

**Comment by markelog** [comment]
> fails the test that you commented(https://github.com/jscs-dev/node-jscs/pull/1405/files#diff-9511adab08ed83a63c8bb84eb0d54617R63). 

Just like i said - this is by design, for example if you do `eslint test.json`, or `jshint test.json` without specifying additional extensions for either of those tools, they will still print the errors, but if you specify `test.json` either in `.jshintignore` or in `.eslintignore` they will be ignored.

We follow common practise here.

**Comment by zxqfox** [comment]
@markelog If file specified in `.someignore` file it should be ignored even if specified as argument. Right? Our `excludeFiles` doesn't skip files if you pass them directly as arguments.

There was an issue about that #1159

**Comment by aliosv** [comment]
@zxqfox, 
>Our excludeFiles doesn't skip files if you pass them directly as arguments.

in my test - skip(https://github.com/aliosv/node-jscs-test/tree/excludeFiles), check please.

**Comment by markelog** [comment]
@zxqfox 

> Our excludeFiles doesn't skip files if you pass them directly as arguments.

Could provide a test-case?

@aliosv Thank you for understanding.

**Pull request closed without merging**